### PR TITLE
chore(tests): adds action tests

### PR DESCRIPTION
**Pull request by hypeJunction**
Adds integration tests to validate that all registered actions are
executable.

Adds a test case abstraction for testing actions

### CODE REVIEW

**Comment by hypeJunction** [comment]
Once this is in I would block any PR that makes changes to any action without adding tests! It's easy.

File: engine/classes/Elgg/Cache/EntityCache.php [hunk 1]
```diff
@@ -89,7 +89,7 @@ public function getByUsername($username) {
 	public function set(ElggEntity $entity) {
 		$guid = $entity->guid;
 
-		if (!$guid || isset($this->entities[$guid]) || isset($this->disabled_guids[$guid])) {
```
Comment (hypeJunction) [inline engine/classes/Elgg/Cache/EntityCache.php]: I don't know why this was here, but it's wrong. We propagate changes after `ElggEntity::update()`

File: engine/classes/Elgg/Database/UsersTable.php [hunk 2]
```diff
@@ -563,12 +563,15 @@ public function setLastAction(ElggUser $user) {
 
 		$this->entity_cache->set($user);
 
+		// Presume that the delayed query will succeed
+		// Otherwise we encounter discrepancies. See #11199
+		$user->storeInPersistedCache(_elgg_get_memcache('new_entity_cache'), $time);
```
Comment (hypeJunction) [inline engine/classes/Elgg/Database/UsersTable.php]: Ref #11199

File: engine/classes/Elgg/Database/UsersTable.php [hunk 3]
```diff
@@ -604,15 +607,12 @@ public function setLastLogin(ElggUser $user) {
...
 
-		// If we save the user to memcache during this request, then we'll end up with the
-		// old (incorrect) attributes cached. Hence we want to invalidate as late as possible.
-		// the user object gets saved
-		register_shutdown_function(function () use ($user) {
-			$user->storeInPersistedCache(_elgg_get_memcache('new_entity_cache'));
-		});
+		// Presume that the delayed query will succeed
+		// Otherwise we encounter discrepancies. See #11199
+		$user->storeInPersistedCache(_elgg_get_memcache('new_entity_cache'));
```
Comment (hypeJunction) [inline engine/classes/Elgg/Database/UsersTable.php]: Ref #11199

File: engine/classes/ElggSession.php [hunk 4]
```diff
@@ -266,6 +267,7 @@ public function getIgnoreAccess() {
 	 */
 	public function setIgnoreAccess($ignore = true) {
 		_elgg_services()->accessCache->clear();
+		_elgg_services()->entityCache->clear();
```
Comment (hypeJunction) [inline engine/classes/ElggSession.php]: Ref #11202

File: engine/classes/ElggUser.php [hunk 5]
```diff
@@ -100,7 +100,11 @@ public function getURL() {
 	 * @return bool
 	 */
 	public function ban($reason = "") {
-		return ban_user($this->guid, $reason);
+		$result = ban_user($this->guid, $reason);
```
Comment (hypeJunction) [inline engine/classes/ElggUser.php]: Refs #11197

File: engine/tests/phpunit/integration/Elgg/Database/SeedingTest.php [hunk 6]
```diff
@@ -60,10 +60,59 @@ public function testCanCreateAdminUser() {
...
+
+		_elgg_invalidate_cache_for_entity($user->guid);
+		_elgg_invalidate_memcache_for_entity($user->guid);
+
+		$this->assertTrue(get_entity($user->guid)->isBanned());
 
 		$user->delete();
 	}
 
+	public function testCanSetUserLanguage() {
```
Comment (hypeJunction) [inline engine/tests/phpunit/integration/Elgg/Database/SeedingTest.php]: Refs #11197

**Comment by hypeJunction** [comment]
FFS you fix one thing, 10 other things break. Adios, Elgg. I am done here.

**Comment by mrclay** [comment]
I'll give this a look later.

**Comment by hypeJunction** [comment]
Migration of attributes to metadata seems to have fixed all the discrapancies in user banned/admin state. Rebased version is coming

**Comment by hypeJunction** [comment]
Replaced by #11309

**Pull request closed without merging**